pkg/install/templates/flux-deployment.yaml.tmpl

```diff
@@ -66,7 +66,9 @@ spec:
         # There are no ":latest" images for flux. Find the most recent
         # release or image version at https://hub.docker.com/r/fluxcd/flux/tags
         # and replace the tag here.
-        image: docker.io/fluxcd/flux:1.18.0
+        image: {{ if .ImageRepository }}{{ .ImageRepository }}flux:1.18.0{{ else }}docker.io/fluxcd/flux:1.18.0{{ end }}{{ if .ImagePullSecrets }}
```

Comment:
I think making the `if`-statement multi-lined would improve readability. If you agree and decide to change this, keep in mind to use the appropriate whitespace controllers (e.g. `{{-`).